crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -1681,6 +1691,44 @@ impl AuthorityPerEpochStore {
         }
     }
 
+    pub(crate) fn notify_settlement_transactions_ready(
+        &self,
+        tx_key: TransactionKey,
+        txns: Vec<VerifiedExecutableTransaction>,
+    ) {
+        debug_assert!(matches!(tx_key, TransactionKey::AccumulatorSettlement(..)));
+        let mut registrations = self.settlement_registrations.lock();
+        if let Some(registration) = registrations.remove(&tx_key) {
+            let SettlementRegistration::Waiting(tx) = registration else {
+                fatal!("Settlement registration should be waiting");
+            };
+            tx.send(txns).unwrap();
```

Comment:
Is there a reason to unwrap here?

Reply:
because the rx is never dropped before it receives a notification, so the send cannot fail

Reply:
rx could be dropped if epoch ends, since it was used in with_live_epoch, and it could happen since it's driven by checkpoint executor, which could in theory went ahead of consensus scheduling? Seems hard to reason about it. Similarly could tx drop first? I noticed we also do `unwrap` on `rx.await`.